Fix #9702  - PHP Warning on exec() in include/utils.php

chris001 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!--- Please be aware that as of the 31st January 2022 we no longer support 7.10.x.
New PRs to hotfix-7.10.x will be invalid. If your fix is still applicable to 7.12.x, 
please create the pull request to the hotfix branch accordingly. -->


## Description
<!--- Describe your changes in detail -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here unless your commit contains the issue number -->
<!--- Ensure that all code ``` is surrounded ``` by triple back quotes. This can also be done over multiple lines -->
Fixes #9702 warnings polluting logs when run on shared hosting with system shell commands disabled.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Also asks user to install the required linux/unix package `php-posix` if not detected.

## How To Test This
<!--- Please describe in detail how to test your changes. -->
Try to run without package `php-posix` should show message saying please install it.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist
<!--- Go over all the following points and check all the boxes that apply. --->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! --->
- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

<!--- Your pull request will be tested via Travis CI to automatically indicate that your changes do not prevent compilation. --->

<!--- If it reports back that there are problems, you can log into the Travis system and check the log report for your pull request to see what the problem was. --->

chris001 committed:
Fix #9702 - PHP Warning on exec() in include/utils.php

- Log error message instead of throw Exception
- Quiet warning exec disabled. Exc Install php-posix

SinergiaCRM commented:
Hi,

This code looks good to us. The error message in the log is quiet annoying. 

Can we include this in the core as soon as possible? @clemente-raposo 

Thanks

SinergiaCRM commented:
Hi @chris001 ,

Perhaps the title of the PR and the commit can be changed to use the standard? https://docs.suitecrm.com/community/contributing-code/bugs/#_providing_a_bug_fix

Fix #ISSUE_NUMBER - Title ....

chris001 commented:
@SinergiaCRM 
Title of PR changed to use the standard! 😃 
Title of the commit is not editable... 😞

SinergiaCRM commented:
Hi @chris001,

To change the last commit, try using `git commit --amend -m "an updated commit message"`.

Regarding the code, be careful with the new exception: until now, processes like [cron.php](https://github.com/salesagility/SuiteCRM/blob/hotfix/cron.php) were always executed, even if the user was not identified or php-posix was not installed. If the exception is thrown and not caught, cron.php would now abort.

chris001 commented:
@SinergiaCRM 
Title of commit message changed to use the standard! 😃
PR updated to log error message, instead of throw possibly uncaught Exception. 😎

jack7anderson7 commented:
Hi @chris001,

Could you please change your base branch to `hotfix` instead of `master`.

Thanks, 
Jack

chris001 committed:
Merge branch 'salesagility:hotfix' into patch-19

chris001 committed:
Fix #9702 - PHP warning on `exec()` in `include/utils.php`

Cleanup error message.

chris001 commented:
@jack7anderson7 Base branch changed to `hotfix` 😎

mapreri commented:
It looks like with PHP 8 a disabled function leads to a `PHP Fatal error: This function has been disabled` exception, exactly on this line.

Could this MR be processed, or what's blocking it?

chris001 committed:
Fix #9702 PHP 7 warning and PHP 8 will exit on error suppressed function call.

Safely determine the running username of PHP process, for file ownership and permissions purposes.
